js/containers/portal-dashboard/answer-compact.tsx

```diff
@@ -1,25 +1,39 @@
 import React from "react";
+import { Map } from "immutable";
 import { getAnswersByQuestion } from "../../selectors/report-tree";
 import { connect } from "react-redux";
 import { getAnswerType, getAnswerIconId, AnswerProps } from "../../util/answer-utils";
+import { feedbackValidForAnswer } from "../../util/misc";
+import { getHideFeedbackBadges } from "../../selectors/dashboard-selectors";
+import QuestionFeedbackBadge from "../../../img/svg-icons/feedback-question-badge-icon.svg";
+import FeedbackAnswerUpdatedBadge from "../../../img/svg-icons/feedback-answer-updated-badge-icon.svg";
 
 import css from "../../../css/portal-dashboard/answer-compact.less";
-
-class AnswerCompact extends React.PureComponent<AnswerProps> {
-  constructor(props: AnswerProps) {
+interface IProps extends AnswerProps {
+  hideFeedbackBadges: boolean;
+  feedback: Map<string, any>;
+}
+class AnswerCompact extends React.PureComponent<IProps> {
+  constructor(props: IProps) {
     super(props);
   }
 
   render() {
-    const { answer, question } = this.props;
+    const { answer, question, feedback, hideFeedbackBadges } = this.props;
     const answerType = getAnswerType(answer, question);
     const iconId = getAnswerIconId(answerType);
+
     return (
       <div className={css.answerCompact} data-cy="student-answer">
         { answer && (!question.get("required") || answer.get("submitted"))
           ? this.renderAnswer(answerType?.icon, iconId)
           : this.renderNoAnswer()
         }
+        {(feedback && !hideFeedbackBadges &&
+          (feedbackValidForAnswer(feedback, answer)
+            ? <QuestionFeedbackBadge className={css.feedbackBadge}/>
+            : <FeedbackAnswerUpdatedBadge className={css.feedbackBadge}/>)
+        )}
```

Comment:
This might be a tad easier to read and more consistent with the code right above it:
```
        { feedback && !hideFeedbackBadges &&
          (feedbackValidForAnswer(feedback, answer)
           ? <QuestionFeedbackBadge className={css.feedbackBadge}/>
           : <FeedbackAnswerUpdatedBadge className={css.feedbackBadge}/>)
        }
```